contracts/RabbitHoleReceipt.sol

```diff
@@ -69,6 +72,12 @@ contract RabbitHoleReceipt is
         royaltyRecipient = royaltyRecipient_;
     }
 
+    /// @dev set the quest factory contract
+    /// @param questFactory_ the address of the quest factory contract
+    function setQuestFactory(address questFactory_) public onlyOwner {
+        QuestFactoryContract = IQuestFactory(questFactory_);
```

Comment:
Is it more common to use Interface here or the actual contract?

Reply:
the actual contract would have been better, but it creates a circular dependency and I couldn't.